pkg/synthetictests/disruption.go

```diff
@@ -33,6 +35,11 @@ func testServerAvailability(
 	backendName := fmt.Sprintf("%s-%s-connections", disruptionName, connType)
 	jobType, err := platformidentification.GetJobType(context.TODO(), restConfig)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
Not found error can be in some cases a transitional error. I don't think `testServerAvailability` takes that into account. Nevertheless, seeing `platformidentification.GetJobType` as a black box, it's not safe to deduce non-existence of a certain API from a NotFoundErr. Worth considering a different approach or to refactore `platformidentification.GetJobType` to provide more output.

Reply:
Changed it to be a bit more robust now. But could that happen? The tests only start after the cluster has been deemed ready, which means all resources and componentes are ready too: https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh#L294-L418